src/main/java/com/splunk/kafka/connect/SplunkSinkRecord.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2017-2018 Splunk, Inc..
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.splunk.kafka.connect;
+
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.kafka.connect.header.Headers;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * SplunkSinkRecord provides helper functionality to enable Header support for the Splunk Connect for Kafka, Namely
+ * Header functionality introspection and comparison.
+ * <p>
+ *
+ * @version     1.1.0
+ * @since       1.1.0
+ */
+public class SplunkSinkRecord {
+    private static final Logger log = LoggerFactory.getLogger(SplunkSinkRecord.class);
+    Headers headers;
+    SplunkSinkConnectorConfig connectorConfig;
+    String splunkHeaderIndex = "";
+    String splunkHeaderHost = "";
+    String splunkHeaderSource = "";
+    String splunkHeaderSourcetype = "";
+
+    public SplunkSinkRecord() {}
+
+    /**
+     * Creates a new Kafka Header utility object. Will take a Kafka SinkRecord and Splunk Sink Connector configuration
+     * and create the object based on Headers included with te Kafka Record.
+     *
+     * @param       record           Kafka SinkRecord to be introspected and headers retrieved from.
+     * @param       connectorConfig  Splunk Connector configuration used to determine headers of importance
+     * @version     1.1.0
+     * @since       1.1.0
+     */
+    public SplunkSinkRecord(SinkRecord record, SplunkSinkConnectorConfig connectorConfig) {
+        this.connectorConfig = connectorConfig;
+        this.headers = record.headers();
+        if(this.headers != null) {
+            log.info("not null headers");
```

Comment:
can we get rid of this log as it will be too many and didn't provide much valuable information

Reply:
haha - yeah :)